### PR TITLE
Add enemy jumping/climbing.

### PR DESCRIPTION
**Pull request by tclaar**
Also add final boss battle, I committed to the wrong branch sorry lol

### CODE REVIEW

**Review by b1gd3vd0g — APPROVED** [review]
Love it

File: js/classes/entities/midground/enemies/enemy_base.js [hunk 1]
```diff
@@ -247,7 +252,10 @@ class EnemyBase {
             // Update bounding box and check collisions.
             this.enemy.lastBoundingBox = this.enemy.boundingBox;
             this.enemy.boundingBox = new BoundingBox(this.enemy.pos, this.enemy.scaledSize);
-            checkBlockCollisions(this.enemy, this.enemy.scaledSize);
+            const collisions = checkBlockCollisions(this.enemy, this.enemy.scaledSize);
```
Comment (calebkrauter) [inline js/classes/entities/midground/enemies/enemy_base.js:255]: Thank you for implementing this! Very clean. Is collisions a good name? Just a thought, idk what it really is. Is it returning a boolean that says if an enemy has collided with a block?

File: js/classes/entities/midground/enemies/wizard.js [hunk 2]
```diff
@@ -106,6 +106,10 @@ class Wizard {
         });
         // Step 5: Now that your position is actually figured out, draw your correct bounding box.
         this.boundingBox = new BoundingBox(this.pos, Wizard.SCALED_SIZE);
+
+        if (ZONE.name === "End Fight Section" && STORY.botsKilled >= 20) {
```
Comment (calebkrauter) [inline js/classes/entities/midground/enemies/wizard.js:110]: Lol hard mode activated.

File: js/classes/entities/midground/portal.js [hunk 3]
```diff
@@ -148,11 +148,11 @@ class Portal {
      * NOTE: consider only spawning new enemies when the previous ones are dead.
```
Comment (calebkrauter) [inline js/classes/entities/midground/portal.js:148]: Good call.

File: js/classes/entities/midground/weapon/water_balloon.js [hunk 4]
```diff
@@ -85,6 +85,14 @@ class WaterBalloon {
         const center = Vector.add(this.pos, Vector.divide(WaterBalloon.SCALED_SIZE, 2));
         GAME.addEntity(new ParticleEffect(center, ParticleEffect.WATER_EXPLOSION));
```
Comment (calebkrauter) [inline js/classes/entities/midground/weapon/water_balloon.js:86]: Love this, nice job @NathanHinthorne

File: js/classes/entities/midground/weapon/water_balloon.js [hunk 5]
```diff
@@ -85,6 +85,14 @@ class WaterBalloon {
...
         ASSET_MGR.playSFX(SFX.WATER_BALLOON.path, SFX.WATER_BALLOON.volume);
+
+        const nearbyEntities = getNearbyEntities(this.pos, 120);
```
Comment (calebkrauter) [inline js/classes/entities/midground/weapon/water_balloon.js:89]: Splash effect? Haha that's great!

File: js/scripts/conversation_loaders/end_conversation_loader.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const none = DialogBubble.SPEAKERS.NONE;
+
+  return {
+      victory: [
+          // 0
+          new DialogBubble(wizard,
+              "What?!? How did you defeat my robot horde?!"),
+          // 1
+          new DialogBubble(chad,
+              "Your puny minions are no match for the power of CHAD, evil wizard!"),
```
Comment (calebkrauter) [inline js/scripts/conversation_loaders/end_conversation_loader.js:19]: XD very nice.

File: js/scripts/conversation_loaders/end_conversation_loader.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+          // 2
+          new DialogBubble(wizard,
+              "I should have known..."),
+          // 3
+          new DialogBubble(wizard,
+            "I wish I had more robots :("),
```
Comment (calebkrauter) [inline js/scripts/conversation_loaders/end_conversation_loader.js:25]: LOL

File: js/scripts/conversation_loaders/end_conversation_loader.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+          // 4
+          new DialogBubble(none,
+            "Congratulations! You defeated the wizard and won the game.")
```
Comment (calebkrauter) [inline js/scripts/conversation_loaders/end_conversation_loader.js:28]: Good. We needed a win situation. Nice work.

File: js/scripts/zone_loaders/end_zone_loader.js
Comment (calebkrauter) [inline js/scripts/zone_loaders/end_zone_loader.js:1]: This is great. Looking forward to seeing this in the caves!

File: js/scripts/zone_loaders/river_zone_loaders.js [hunk 9]
```diff
@@ -30,20 +30,21 @@ const loadRiver = () => {
         // NPCs
         ASSET_MGR.queueDownload(Bird.SPRITESHEET);
 
+        ASSET_MGR.queueDownload(MUSIC.PEACEFUL_CHIPTUNE.path);
     };
 
     const addEntities = () => {
         // Add a border to the right side of the map
-        GAME.addEntity(new Border(
```
Comment (calebkrauter) [inline js/scripts/zone_loaders/river_zone_loaders.js:37]: Why is this removed?

**Review by calebkrauter — APPROVED** [review]
Looks good. Nice work.
